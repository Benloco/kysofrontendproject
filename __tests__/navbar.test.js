/**
 * @jest-environment jsdom
 */

import React, { Component } from 'react'
import { render, screen,cleanup,fireEvent } from '@testing-library/react'
import Navbar from '../pages/navbar/navbar'
import renderer from 'react-test-renderer';

afterEach(cleanup);

describe('Navbar',()=>{
    it("renders navbar without crashing",()=>{
        render(<Navbar/>)
    })

    it("renders the project title",()=>{
        render(<Navbar/>);
        const value=   screen.getByText("Kyso Frontend Project");
        expect(value).toBeInTheDocument();
    })

    it("has the project title text as is",()=>{
        render(<Navbar/>);
        const value=   screen.getByText("Kyso Frontend Project");
        expect(value).toHaveTextContent("Kyso Frontend Project");
        fireEvent.click(value);
    })
    
    it("renders navbar element (Dashboard)",()=>{
        render(<Navbar/>);
        const value=   screen.getByText("Dashboard");
        expect(value).toHaveTextContent("Dashboard");
        fireEvent.click(value);
    })

    it("matches snapshot",()=>{
       const navtree= renderer.create(<Navbar/>).toJSON();
       expect(navtree).toMatchSnapshot();
    })


})