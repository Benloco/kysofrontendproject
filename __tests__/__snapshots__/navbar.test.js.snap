// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar matches snapshot 1`] = `
<nav
  className="navbar navbar-dark bg-dark fixed-top"
  style={
    Object {
      "zIndex": 100,
    }
  }
>
  <a
    className="navbar-brand ml-3"
    href="/"
  >
    Kyso Frontend Project
  </a>
  <a
    className="navbar-brand"
    href="/dashboard"
  >
    Dashboard
  </a>
</nav>
`;
